presto-kudu/src/main/java/com/facebook/presto/kudu/KuduClientSession.java

```diff
@@ -0,0 +1,594 @@
+
+    public Schema getTableSchema(KuduTableHandle tableHandle)
+    {
+        KuduTable table = tableHandle.getTable(this);
+        return table.getSchema();
+    }
+
+    public Map<String, Object> getTableProperties(KuduTableHandle tableHandle)
+    {
+        KuduTable table = tableHandle.getTable(this);
+        return KuduTableProperties.toMap(table);
+    }
+
+    public List<KuduSplit> buildKuduSplits(KuduTableLayoutHandle layoutHandle)
+    {
+        KuduTableHandle tableHandle = layoutHandle.getTableHandle();
+        KuduTable table = tableHandle.getTable(this);
+        final int primaryKeyColumnCount = table.getSchema().getPrimaryKeyColumnCount();
+        KuduScanToken.KuduScanTokenBuilder builder = client.newScanTokenBuilder(table);
+
+        TupleDomain<ColumnHandle> constraintSummary = layoutHandle.getConstraintSummary();
+        if (!addConstraintPredicates(table, builder, constraintSummary)) {
+            return ImmutableList.of();
+        }
+
+        Optional<Set<ColumnHandle>> desiredColumns = layoutHandle.getDesiredColumns();
+        if (desiredColumns.isPresent()) {
+            if (desiredColumns.get().contains(KuduColumnHandle.ROW_ID_HANDLE)) {
+                List<Integer> columnIndexes = IntStream
+                        .range(0, primaryKeyColumnCount)
+                        .boxed().collect(Collectors.toList());
+                for (ColumnHandle columnHandle : desiredColumns.get()) {
+                    if (columnHandle instanceof KuduColumnHandle) {
+                        KuduColumnHandle k = (KuduColumnHandle) columnHandle;
+                        int index = k.getOrdinalPosition();
+                        if (index >= primaryKeyColumnCount) {
+                            columnIndexes.add(index);
+                        }
+                    }
+                }
+                builder.setProjectedColumnIndexes(columnIndexes);
+            }
+            else {
+                List<Integer> columnIndexes = desiredColumns.get().stream()
+                        .map(handle -> ((KuduColumnHandle) handle).getOrdinalPosition())
+                        .collect(toImmutableList());
+                builder.setProjectedColumnIndexes(columnIndexes);
+            }
+        }
+
+        List<KuduScanToken> tokens = builder.build();
+        return tokens.stream()
+                .map(token -> toKuduSplit(tableHandle, token, primaryKeyColumnCount))
+                .collect(toImmutableList());
+    }
+
+    public KuduScanner createScanner(KuduSplit kuduSplit)
+    {
+        try {
+            return KuduScanToken.deserializeIntoScanner(kuduSplit.getSerializedScanToken(), client);
+        }
+        catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Converts to raw name and opens table.
+     *
+     * @param schemaTableName table name
+     * @return KuduTable
+     * @throws SchemaNotFoundException
+     * @throws TableNotFoundException
```

Comment:
Comments like below do not add any value to the code:

```
     *
      * @param schemaTableName table name
      * @return KuduTable
      * @throws SchemaNotFoundException
      * @throws TableNotFoundException
```

Please try to avoid them.